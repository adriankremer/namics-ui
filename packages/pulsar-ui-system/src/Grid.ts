import { cx, css } from "emotion";
import { ToolbarProps } from "@pulsar-ui/core";
import { PulsarBoxProps } from "./Box";

export function useGridProps(
  _: PulsarBoxProps,
  { className, ...htmlProps }: ToolbarProps
) {
  const grid = css`
    display: grid;
    grid-gap: 15px 20px;
    grid-template-columns: repeat(12, 1fr);
  `;
  return { ...htmlProps, className: cx(grid, className) };
}
