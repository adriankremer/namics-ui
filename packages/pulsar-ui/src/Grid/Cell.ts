import use from "reuse";
import Box, { BoxProps } from "../Box";
import styled from "../styled";

interface CellProps extends BoxProps {
  width: number;
  height: number;
}

const Cell = styled(Box)<CellProps>(props => props.theme.Cell);

export default use(Cell, "div");
