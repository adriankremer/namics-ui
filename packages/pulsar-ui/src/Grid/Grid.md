Grid example:

```js
<Grid>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>1/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>2/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>3/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>4/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>5/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>6/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>7/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>8/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>9/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>10/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>11/12</div>
  </Cell>
  <Cell width="1">
    <div style={{height: "100%", background: "gray"}}>12/12</div>
  </Cell>

  <Cell width="2">
    <div style={{height: "100%", background: "gray"}}>2/12</div>
  </Cell>
</Grid>
```
