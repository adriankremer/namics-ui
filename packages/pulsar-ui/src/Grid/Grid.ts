import use from "reuse";
import Box, { BoxProps } from "../Box";
import styled from "../styled";

export interface IGridProps extends BoxProps {
  fill?: string;
}

const Grid = styled(Box)<IGridProps>`
  display: grid;
  /* @todo: we need utils for this.. */
  grid-gap: ${props =>
    (props.theme.spacing && props.theme.spacing[1]) || ".5rem"};
  grid-template-columns: repeat(12, 1fr);
  ${props => props.theme.Grid}
`;

export default use(Grid, "div");
