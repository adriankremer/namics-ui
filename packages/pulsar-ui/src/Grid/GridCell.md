```js
<Grid>
  <Grid.Cell width={6} height={4}>
    <Avatar width="100%" src="https://placekitten.com/400/400" />
  </Grid.Cell>
  <Grid.Cell width={3} height={1}>
    <Avatar width="100%" src="https://placekitten.com/300/300" />
  </Grid.Cell>
  <Grid.Cell width={1} height={1}>
    <Avatar width="100%" src="https://placekitten.com/150/150" />
  </Grid.Cell>
</Grid>
```
