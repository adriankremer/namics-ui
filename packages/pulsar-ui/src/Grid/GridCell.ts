import use from "reuse";
import { theme, prop, ifProp } from "styled-tools";
import Box, { BoxProps } from "../Box";
import styled, { css } from "../styled";

export interface CellProps extends BoxProps {
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  width?: number;
  height?: number;
}

const GridCell = styled(Box)<CellProps>`
  grid-column-end: span ${prop("width", "12")};
  grid-row-end: span ${prop("height", "1")};

  @media screen and (min-width: ${theme("breakpoints.sm", "480px")}) {
    ${ifProp("sm", `grid-column-end: span ${prop("sm")};`)}
  }
  @media screen and (min-width: ${theme("breakpoints.md", "768px")}) {
    ${ifProp("md", `grid-column-end: span ${prop("md")};`)}
  }
  @media screen and (min-width: ${theme("breakpoints.lg", "1024px")}) {
    ${ifProp("lg", `grid-column-end: span ${prop("lg")};`)}
  }
  @media screen and (min-width: ${theme("breakpoints.xl", "1200px")}) {
    ${ifProp("xl", `grid-column-end: span ${prop("xl")};`)}
  }
  ${theme("GridCell")}
`;

export default use(GridCell, "div");
