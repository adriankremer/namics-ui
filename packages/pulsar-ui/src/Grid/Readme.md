---
title: "Grid"
path: "/components/grid"
---

```jsx
import { Grid, GridCell, Card } from "@pulsar-ui/core";
import { Avatar } from "@pulsar-ui/core";

function Example() {
  return (
    <Grid>
      <GridCell system={{cellWidth: [12, 4, 4, 3]}}>
        <Avatar src="https://placekitten.com/400/400" width="100%" />
      </GridCell>
      <GridCell system={{cellWidth: [12, 8, 8, 9]}}>
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.
        </p>
      </GridCell>
      <GridCell system={{cellWidth: [12,4]}}>
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
        </p>
      </GridCell>
      <GridCell system={{cellWidth: [12,4]}}>
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
        </p>
      </GridCell>
      <GridCell system={{cellWidth: [12,4]}}>
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
        </p>
      </GridCell>
    </Grid>
  )
}
```
