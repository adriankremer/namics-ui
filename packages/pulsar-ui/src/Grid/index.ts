import Grid from "./Grid";
import GridCell from "./GridCell";

export * from "./Grid";
export * from "./GridCell";

export default Object.assign(Grid, {
  Cell: GridCell
});
