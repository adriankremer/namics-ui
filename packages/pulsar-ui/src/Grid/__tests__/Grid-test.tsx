import * as React from "react";
import { render } from "react-testing-library";
import Grid from "../Grid";

test("html, attr", () => {
  const { getByTestId } = render(
    <Grid id="grid" aria-label="test" data-testid="test" />
  );
  expect(getByTestId("test")).toHaveAttribute("id", "grid");
  expect(getByTestId("test")).toHaveAttribute("aria-label", "test");
});

test("styles", () => {
  const { container } = render(<Grid />);
  expect(container.firstChild).toMatchSnapshot();
});
