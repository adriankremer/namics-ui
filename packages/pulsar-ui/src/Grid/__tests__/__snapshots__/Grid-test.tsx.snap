// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  display: grid;
  grid-gap: .5rem;
  grid-template-columns: repeat(12,1fr);
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0"
/>
`;
